File: tutorials/testing/java/tut_worldmap_p2.java
import java.util.ArrayList;
import java.util.Map;
import bridges.connect.Bridges;
import bridges.connect.DataSource;
import bridges.data_src_dependent.Country;
import bridges.base.WorldMap;
import bridges.base.Color;


// This tutorial illustrates how to access the data of the World map with 
// boundaries; the countries are filled with default color
public class tut_worldmap_p2 {
	public static void main(String[] args) throws Exception {

		// create Bridges object
		Bridges bridges = new Bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", 
												"YOUR_API_KEY");
		// set title, description
		bridges.setTitle("Tutorial: Drawing US Map with State Boundaries");
		bridges.setDescription("All US states, No Counties");

		// get data source object
		DataSource ds = bridges.getDataSource();

		// specify a few countries to change their fill colors
		String[] countries = {"Brazil", "Argentina", "Germany", "France", "Spain", 
									"United States of America"};
		// get us map data - states and boundaries
		ArrayList<Country> wm_data = ds.getWorldMapData (countries);

		// lets change colors of each country
		wm_data.get(0).setFillColor(new Color("lightgoldenrodyellow"));
		wm_data.get(1).setFillColor(new Color("lightskyblue"));
		wm_data.get(2).setFillColor(new Color("indigo"));
		wm_data.get(3).setFillColor(new Color("orangered"));
		wm_data.get(4).setFillColor(new Color("olive"));
		wm_data.get(5).setFillColor(new Color("springgreen"));

		WorldMap wm = new WorldMap(wm_data);

		// set data structure
		bridges.setDataStructure(wm);

		// visualize
		bridges.visualize();
	}
}

File: tutorials/testing/c++/tut_worldmap_p2.cpp
#include <string>

using namespace std;

#include "Bridges.h"
#include "DataSource.h"
#include "WorldMap.h"

using namespace bridges;

int main(int argc, char **argv) {

	// create bridges object
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");

	// set title
	bridges.setTitle("Drawing a World Map");

	// set description
	bridges.setDescription("This example illustrates how to draw a world map of a few countries");

	// create a data source
	DataSource ds(&bridges);

	// specify a few countries to draw their maps
	vector<string> countries = {"Brazil", "Argentina", "Germany", "France", "Spain", 
							"United States of America"};
	vector<Country>  wm_data = ds.getWorldMapData({countries});

	// lets change colors of each country
	wm_data[0].setFillColor(Color("lightgoldenrodyellow"));
	wm_data[1].setFillColor(Color("lightskyblue"));
	wm_data[2].setFillColor(Color("indigo"));
	wm_data[3].setFillColor(Color("orangered"));
	wm_data[4].setFillColor(Color("olive"));
	wm_data[5].setFillColor(Color("springgreen"));
	
	WorldMap wm (wm_data);
	
	bridges.setDataStructure(&wm);
	bridges.visualize();
	
	return 0;
}
